Framework/API/src/FunctionFactory.cpp

```diff
@@ -279,10 +297,28 @@ void FunctionFactoryImpl::addConstraints(IFunction_sptr fun,
  * @param fun :: The function
  * @param expr :: The constraint expression.
  */
-void FunctionFactoryImpl::addConstraint(IFunction_sptr fun,
+void FunctionFactoryImpl::addConstraint(boost::shared_ptr<IFunction> fun,
```

Comment:
I changed `IFunction_sptr` to `boost::shared_ptr<IFunction>` because overloading `FunctionFactoryImpl::addConstraint` caused doxygen to fail using `IFunction_sptr`.  I think there are three options, and I'm not sure which is best, so let me know which would be appropriate:

1) We can update the doxygen build server to Ubuntu 18.04 (https://mantid.slack.com/archives/C02J3KYQ1/p1538496438000100)
2) Not overload the function (call the function using a non-default penalty something like `FunctionFactoryImpl::addConstraintWithNonDefaultPenalty()`
3) Leave it like this (at least until the doxygen build server is eventually upgraded)

Reply:
I believe 3 is best at this moment.